frontend/src/posapp/components/pos/Payments.vue

```
@@ -957,21 +985,113 @@ const clearBackgroundStatusCheck = () => {
-const scheduleBackgroundStatusCheck = (invoiceName, doctype) => {
+const resolveSubmittedDoctype = (doctype) => {
+	if (doctype) return doctype;
+	if (invoice_doc.value?.doctype) return invoice_doc.value.doctype;
+	return pos_profile.value?.create_pos_invoice_instead_of_sales_invoice
+		? "POS Invoice"
+		: "Sales Invoice";
+};
+
+const fetchSubmittedInvoiceDoc = async (invoiceName, doctype) => {
+	const resolvedDoctype = resolveSubmittedDoctype(doctype);
+	return frappe.db.get_doc(resolvedDoctype, invoiceName);
```

ESLint

[error] 'frappe' is not defined.

(no-undef)